src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/servlet/CompleteServletAnnotationQuickFix.java

```diff
@@ -66,69 +77,80 @@ protected void insertAnnotations(Diagnostic diagnostic, JavaCodeActionContext co
         }
     }
 
-    private static void insertAndReplaceAnnotation(Diagnostic diagnostic, JavaCodeActionContext context,
-                                                   List<CodeAction> codeActions, String annotation) {
-
-        // Insert the annotation and the proper import by using JDT Core Manipulation
-        // API
-
-        // if missing an attribute, do value insertion
+    @Override
+    public CodeAction resolveCodeAction(JavaCodeActionResolveContext context) {
+        final CodeAction toResolve = context.getUnresolved();
         PsiElement node = null;
         PsiModifierListOwner parentType = null;
         PsiAnnotation annotationNode = null;
+        CodeActionResolveData data = (CodeActionResolveData) toResolve.getData();
+        String diagnosticCode = (String) data.getExtendedDataEntry(DIAGNOSTIC_CODE_KEY);
+        String attribute = (String) data.getExtendedDataEntry(ATTRIBUTE_KEY);
+        String annotation = (String) data.getExtendedDataEntry(ANNOTATION_KEY);
+
+        if (diagnosticCode.equals(ServletConstants.DIAGNOSTIC_CODE_MISSING_ATTRIBUTE)) {
+            node = context.getCoveringNode();
+            parentType = getBinding(node);
+            annotationNode = getAnnotation(node);
+
+            ArrayList<String> attributesToAdd = new ArrayList<>();
+            attributesToAdd.add(attribute);
+            String name = getLabel(annotation, attribute, "Add");
```

Comment:
Should retrieve the title from the code action instead of recomputing it here.

Reply:
Made the necessary changes so that the title is retrieved from `toResolve.getTitle()`.